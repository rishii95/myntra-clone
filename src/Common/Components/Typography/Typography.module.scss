@import "../../variables.module.scss";

.subtitle {
    font-size: $subtitle-size;
}

.text {
    font-size: $text-size;
}
