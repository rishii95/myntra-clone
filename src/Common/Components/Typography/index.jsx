/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { PropTypes } from 'prop-types';
import { Typography as TypographyAnt } from 'antd';

import variables from '../../variables.module.scss';

import TypographyStyles from './Typography.module.scss';

const { Title, Paragraph, Text } = TypographyAnt;

export default function Typography({
  children, type, color, weight, className, marginBottom, ...props
}) {
  const getStyles = () => ({ marginBottom, color, fontWeight: weight });
  const getElement = () => {
    switch (type) {
      case 'title':
        return (
          <Title
            level={5}
            style={{ ...getStyles(), fontWeight: variables.boldFont }}
            className={className}
            {...props}
          >
            {children}
          </Title>
        );
      case 'subtitle':
        return (
          <Paragraph
            className={`${TypographyStyles.subtitle} ${className}`}
            style={getStyles()}
            {...props}
          >
            {children}
          </Paragraph>
        );
      case 'text':
        return (
          <Text
            className={`${TypographyStyles.text} ${className}`}
            style={getStyles()}
            {...props}
          >
            {children}
          </Text>
        );
      default:
        return null;
    }
  };

  return getElement();
}

Typography.propTypes = {
  children: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['title', 'text', 'subtitle']).isRequired,
  color: PropTypes.string,
  weight: PropTypes.string,
  className: PropTypes.string,
  marginBottom: PropTypes.number,
};

Typography.defaultProps = {
  color: variables.black,
  weight: variables.regularFont,
  className: '',
  marginBottom: 0,
};
