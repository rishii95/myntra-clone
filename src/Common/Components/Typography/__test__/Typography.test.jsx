/* eslint-disable no-undef */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Typography from '../index';

afterEach(cleanup);
test('it matches snapshot', () => {
  const { container } = render(<Typography type="title">Title</Typography>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders title correctly', () => {
  const { getByTestId } = render(<Typography type="title">Title</Typography>);
  expect(getByTestId('typography-title')).toHaveTextContent('Title');
});

test('renders subtitle correctly', () => {
  const { getByTestId } = render(<Typography type="subtitle">Subtitle</Typography>);
  expect(getByTestId('typography-subtitle')).toHaveTextContent('Subtitle');
});

test('renders text correctly', () => {
  const { getByTestId } = render(<Typography type="text">Text</Typography>);
  expect(getByTestId('typography-text')).toHaveTextContent('Text');
});
