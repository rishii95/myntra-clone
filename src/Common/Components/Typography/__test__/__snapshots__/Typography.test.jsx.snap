// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<h5
  class="ant-typography"
  data-testid="typography-title"
  style="margin-bottom: 0px; color: black; font-weight: boldFont;"
>
  Title
</h5>
`;
